modules/apps/data-engine/data-engine-taglib/test/js/data-layout-builder/components/rule-builder/editor/Calculator.es.js

```diff
@@ -0,0 +1,377 @@
+	{
+		dataType: 'integer',
+		fieldName: 'number2',
+		label: 'number2',
+		name: 'number2',
+		options: [],
+		repeatable: false,
+		title: 'number2',
+		type: 'numeric',
+		value: '',
+	},
+	{
+		dataType: 'integer',
+		fieldName: 'total',
+		label: 'total',
+		name: 'total',
+		options: [],
+		repeatable: false,
+		title: 'total',
+		type: 'numeric',
+		value: '',
+	},
+];
+
+const defaultProps = (fieldsList = fields) => {
+	return {
+		expression: '',
+		fields: fieldsList,
+		functions: [
+			{
+				label: 'sum',
+				tooltip: '',
+				value: 'sum',
+			},
+		],
+		index: 0,
+	};
+};
+
+const changeActionObject = (newExpression) => ({
+	payload: {
+		value: newExpression,
+	},
+	type: 'CHANGE_ACTION_CALCULATE',
+});
+
+describe('Calculator', () => {
+	afterEach(() => {
+		cleanup();
+		jest.restoreAllMocks();
+	});
+
+	describe("clicking the calculator's signals, numbers and parethesis", () => {
+		it('mounts an expression with numbers, signals and parenthesis', () => {
+			const mockOnChange = jest.fn();
+			const props = defaultProps();
+			const {getByText} = render(
+				<Calculator
+					{...props}
+					onChange={(newExpression) =>
+						mockOnChange(changeActionObject(newExpression))
+					}
+				/>
+			);
+
+			fireEvent.click(getByText('('));
+			fireEvent.click(getByText('3'));
+			fireEvent.click(getByText('+'));
+			fireEvent.click(getByText('5'));
+			fireEvent.click(getByText(')'));
+
+			expect(mockOnChange.mock.calls).toHaveLength(5);
+
+			const values = ['(', '(3', '(3+', '(3+5', '(3+5)'];
+
+			mockOnChange.mock.calls.forEach((call, i) =>
+				expect(call).toEqual([changeActionObject(values[i])])
+			);
+		});
+
+		it('does not allow to add consecutive operators', () => {
+			const mockOnChange = jest.fn();
+			const {getByText} = render(
+				<Calculator
+					{...defaultProps()}
+					onChange={(newExpression) =>
+						mockOnChange(changeActionObject(newExpression))
+					}
+				/>
+			);
+
+			let button = getByText('+');
+			fireEvent.click(button);
+			button = getByText('-');
+			fireEvent.click(button);
+			button = getByText('*');
+			fireEvent.click(button);
+			button = getByText('/');
+			fireEvent.click(button);
+
+			expect(mockOnChange.mock.calls).toHaveLength(1);
+
+			expect(mockOnChange.mock.calls).toEqual([
+				[changeActionObject('+')],
+			]);
+		});
+
+		it('disables operators when last token is an operator', () => {
+			const mockOnChange = jest.fn();
+			const {getByText} = render(
+				<Calculator
+					{...defaultProps()}
+					expression="3+3"
+					onChange={(newExpression) =>
+						mockOnChange(changeActionObject(newExpression))
+					}
+				/>
+			);
+
+			fireEvent.click(getByText('+'));
+
+			const symbolsArray = SIGNALS.concat(DOT);
+
+			symbolsArray.forEach((symbol) => {
+				expect(getByText(symbol).disabled).toBe(true);
+			});
+		});
+	});
+
+	describe('clicking the Sum function', () => {
+		it('disables functions, numbers and operators', () => {
+			const mockOnChange = jest.fn();
```

Comment:
Can We avoid repetitions of defining `mockOnChange` moving this definition outside `describe` and then reuse in each test case?

Reply:
Sure! I've this new pr https://github.com/liferay-forms/liferay-portal/pull/91 doing this. Thanks, Dieguito. And welcome back!!